src/types/component.rs

```diff
@@ -137,21 +167,41 @@ pub struct Text {
     pub id: Option<ComponentId>,
     pub text: Arc<str>,
 
+    /// Width of a texture that text will be rendered on. If not provided the resulting texture
+    /// will be sized based on the defined text, but limited to `max_width` value.
```

Comment:
Please clarify, no key is required here, all combinations are supported except very specific one (unknown width and known height).
And I would argue that one unsupported combination does not really make sense.

Reply:
Good point, I must have copied it from weight/height and did not notice. The sentence "Width of a texture that text will be rendered on" for max_width was incorrect.

> Based solely on the max_width description one might understand that in the case of specifying max_width output always has output texture width of max_width.

In this case, max_width had incorrect information so it's fair, but I disagree with the take that field like max_width needs to make sense on its own if a user does not read any other field's description.

> /// If text is longer than specified `max_width`, it will be wrapped accordingly
> /// to the `wrap` parameter.

- max width limits width of a component
- width explicitly defines width of a component
- warp defines a behavior based on that width
I don't think we should mention all fields that will be impacted by that field in the description